pkg/controller/job/job_controller.go

```diff
@@ -759,7 +776,8 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (rErr error) {
 		job.Status.UncountedTerminatedPods = &batch.UncountedTerminatedPods{}
 	}
 	uncounted := newUncountedTerminatedPods(*job.Status.UncountedTerminatedPods)
-	expectedRmFinalizers := jm.finalizerExpectations.getExpectedUIDs(key)
+	syncJobContext.uncounted = uncounted
+	syncJobContext.expectedRmFinalizers = jm.finalizerExpectations.getExpectedUIDs(key)
```

Comment:
it looks like we could move this up when we first create `syncJobCtx`

Reply:
I have actually moved down the creation of `syncJobCtx`